pytube/parser.py

```diff
@@ -0,0 +1,83 @@
+import ast
+import json
+import re
+from pytube.exceptions import HTMLParseError
+
+
+def parse_for_object(html, preceding_regex):
+    """Parses input html to find the end of a JavaScript object.
```

Comment:
can we add where we have a need to parse the object? 
if i remember correctly it's because youtube literally sends javascript as html so you need to parse it out?

Reply:
Yeah, we're parsing the javascript portion of the html page for a video to extract information. In some cases, the javascript sets variables a long, descriptive objects, and we *were* using simple regex patterns to try and find these, but that becomes... problematic pretty quickly. I gave some examples where a normal regex fails to accurately capture the object because it terminates too early, and python regexes don't support overlapping matches (meaning we can't use `findall()` and iterate through results until we find a correct match).